pre_commit_hooks/make/phony-targets.sh

```diff
@@ -5,9 +5,13 @@ set -e
 EXITCODE=0
 
 validate_phony_targets() {
-    local file=$1
-    local source=$(make -npq -f "${file}" | grep -E '^.PHONY:')
-    local targets=$(sed -n 's#^.PHONY:\(.*\)#\1#p' <<< "${source}")
+    local file
+    local source
+    local targets
+
+    file=$1
```

Comment:
Not needed to be split in single line.. parameter assignment should happen more readable in the first lines of the function..

Reply:
shellchecker is suggesting the following:

```
In pre_commit_hooks/make/phony-targets.sh line 9:
    local source=$(make -npq -f "${file}" | grep -E '^.PHONY:')
          ^----^ SC2155: Declare and assign separately to avoid masking return values.

In pre_commit_hooks/make/phony-targets.sh line 10:
    local targets=$(sed -n 's#^.PHONY:\(.*\)#\1#p' <<< "${source}")
          ^-----^ SC2155: Declare and assign separately to avoid masking return values.
```

Reply:
read more about that hint here https://github.com/koalaman/shellcheck/wiki/SC2155

> The exit status of the command is overridden by the exit status of the creation of the local variable.

Reply:
```
validate_phony_targets() {
    local file=$1
    local source
    local targets

    source=$(make -npq -f "${file}" | grep -E '^.PHONY:')
    targets=$(sed -n 's#^.PHONY:\(.*\)#\1#p' <<<"${source}")

    for target in $targets; do
        # The q command for setting exit codes if a pattern doesn't match,
        # isn't supported in OSX / BSD sed.
        # We check the exact match as a workaround instead.
        # If the pattern doesn't match that means that the target doesn't exist.
        if test -z "$(sed -ne "s#^\(${target}:\).*#\1#p" "${file}")"; then
            echo "No target found for PHONY target: '$target'"
            EXITCODE=1
        fi
    done
}
```

Reply:
that's what i meant ;) thanks ..